HighwayPatrol.cpp

```diff
@@ -28,9 +32,17 @@ void HighwayPatrol::pullOver( Vehicle* v, bool willArrest, Highway* h )
     std::cout << name << ": vehicle is traveling " << v->speed - h->speedLimit << " miles per hour over the speed limit" << std::endl;
     if( willArrest )
     {
-        assert(false);
+        //assert(false);
         //print the vehicle type in this std::cout between "THE [" and "] PULL". 
-        std::cout << name << ": YOU IN THE [ " << " ] PULL OVER AND SHOW YOUR HANDS" << std::endl;
+        if ( Motorcycle* bike = dynamic_cast<Motorcycle*>(v) ) //bike->lanesplitAndRace();
+            std::cout << name << ": YOU IN THE [ bike ] PULL OVER AND SHOW YOUR HANDS" << std::endl;
+
+        if ( Car* car = dynamic_cast<Car*>(v) )
+            std::cout << name << ": YOU IN THE [ car ] PULL OVER AND SHOW YOUR HANDS" << std::endl;
+
+        if ( SemiTruck* truck = dynamic_cast<SemiTruck*>(v) )
+            std::cout << name << ": YOU IN THE [ truck ] PULL OVER AND SHOW YOUR HANDS" << std::endl;
```

Comment:
D.R.Y. this.
Don't forget to use `else-if` when figuring out the type. 
create a `std::string` variable to store the type name.
print that out in the `std::cout` statement you commented out below. 
- [ ] revise